src/client/dfuse/il/int_posix.c

```diff
@@ -2886,5 +2886,12 @@ dfuse_get_bypass_status(int fd)
 	return rc;
 }
 
+DFUSE_PUBLIC void __attribute__ ((__noreturn__))
+dfuse__exit(int rc)
+{
+	ioil_fini();
```

Comment:
have you tested this properly? this can disconnect containers and pools that are on the main process i believe.
If I understand this correctly, this can call pool_disconnect for example on a pool, close it on the server, and when the main thread tries to access the pool, the server complains that the pool is not connected. but im not 100% sure on this.

unfortunately we do not have a lot of IL tests in CI with things like FIO and others.

Reply:
https://man7.org/linux/man-pages/man2/exit.2.html
"_exit()" is called to terminate the current process (including all threads). I think it is appropriate to call "ioil_fini()". "pthread_exit()" is the function for thread call to terminate current thread.

Reply:
maybe you did not get my point.
you start with 1 process, it calls init, connect to pool A, container B.
fork();
let's say child does nothing and and calls _exit();
child calls ioil_fini() and closes pool handle A and container handle B
now for the client side, those handles are different, but for the server side, they are the same. so those handles are closed on the server.

parent tries to access pool A and gets an error.

Reply:
ah. I see. If this is an issue, we should have same issue if one process (either parent or child process) exits without calling "_exit" since destructor function "ioil_fini" will be invoked.

Reply:
no. because of this:
https://github.com/daos-stack/daos/blob/master/src/client/dfuse/il/int_posix.c#L369

your patch is actually a no-op because you also hit that condition and fini from those forked processes are not doing anything :-)

Reply:
Good to know. Thank you!
If the parent process calls "_exit()", the destructor function will not be called.
ok. I will revise the PR to only call daos_eq_destroy() to clean up EQs.

Reply:
that will also not work. because for threads, the eq pool is now shared and even if a thread _exit() the eq it had created will be destroyed by the parent when it exits.
so you will need to be able differentiate a new thread calling _exit(); and a child process from fork() calling _exit() and destroy only on the latter.